packages/x-charts/src/internals/store/useCharts.ts

```diff
@@ -94,11 +92,10 @@ export function useCharts<
       svgRef: innerSvgRef,
       chartRootRef: innerChartRootRef,
       seriesConfig,
-      models,
     });
 
     if (pluginResponse.publicAPI) {
-      Object.assign(publicAPI, pluginResponse.publicAPI);
+      Object.assign(publicAPI.current, pluginResponse.publicAPI);
```

Comment:
Is it safe to access a ref during render? If I understand correctly, [the docs](https://react.dev/learn/referencing-values-with-refs#recap) disincentivize it:

> Don’t read or write ref.current during rendering. This makes your component hard to predict.

Reply:
I think this comment is about something like `ref.current === 'renderA' ? <A /> : <B />`

Here we are updating the ref such that devs get access to the setters.